### PR TITLE
feat: linked list exploration

### PR DESCRIPTION
**Pull request by aidenybai**
thoughts? @SukkaW @QuiiBz performance benchmarks are pretty inconsistent for me, not sure whether this is faster than v2.1.0 or not.

Linked list traversal seems to be ~2x faster than array traversal

### CODE REVIEW

**Comment by cytronicoder** [comment]
Oooh, interesting - wouldn't storing elements contiguously in memory be faster than pointers?

**Comment by aidenybai** [comment]
Haven't benchmarked that, but that's a good point

File: packages/million/dom.ts [hunk 1]
```diff
@@ -86,7 +86,7 @@ export const createEventListener = (
         el = el.parentNode;
       }
     });
-    document[EVENTS_REGISTRY].add(event);
```
Comment (SukkaW) [inline packages/million/dom.ts:89]: AFAIK, the polymorphic object is slow (per hidden class).

Do we have any benchmark for this change?

File: packages/million/types.ts [hunk 2]
```diff
@@ -34,113 +34,129 @@ export abstract class AbstractBlock {
...
-  PATCH,
-  BLOCK,
+export const enum Flags {
+  CHILD = 1,
+  EVENT = 2,
+  ATTRIBUTE = 4,
+  STYLE_ATTRIBUTE = 8,
+  SVG_ATTRIBUTE = 16,
+  BLOCK = 32,
 }
```
Comment (SukkaW) [inline packages/million/types.ts:44]: Does esbuild support cross-file inline const enum now? If not, using `export const enum` will increase the bundle size and decrease the performance.

**Comment by aidenybai** [comment]
interesting stuff, but slower than normal object array

**Pull request closed without merging**